examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,223 @@
+import type { ManagedCollection, ManagedCollectionField } from "framer-plugin"
+import type { DataSource, FieldIds, ManagedCollectionFields } from "./data"
+
+import { framer } from "framer-plugin"
+import { useState, useMemo, useEffect, memo } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields, syncCollection } from "./data"
+import { Spinner } from "./components/Spinner"
+
+function ChevronIcon() {
+    return (
+        <svg xmlns="http://www.w3.org/2000/svg" width="8" height="16">
+            <path
+                d="M 3 11 L 6 8 L 3 5"
+                fill="transparent"
+                strokeWidth="1.5"
+                stroke="#999"
+                strokeLinecap="round"
+                strokeLinejoin="round"
+            />
+        </svg>
+    )
+}
+
+interface FieldMappingRowProps {
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+const FieldMappingRow = memo(({ field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) => {
+    return (
+        <>
+            <div
+                className="source-field"
+                aria-disabled={isIgnored}
+                onClick={() => onFieldToggle(field.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    checked={!isIgnored}
+                    onChange={event => {
+                        event.stopPropagation()
+                    }}
+                />
+                <span>{field.id}</span>
+            </div>
+            <ChevronIcon />
+            <input
+                type="text"
+                style={{
+                    width: "100%",
+                    opacity: isIgnored ? 0.5 : 1,
+                }}
+                disabled={isIgnored}
+                placeholder={field.id}
+                value={field.name}
+                onChange={event => onFieldNameChange(field.id, event.target.value)}
+                onKeyDown={event => {
+                    if (event.key === "Enter") {
+                        event.preventDefault()
+                    }
+                }}
+            />
+        </>
+    )
+})
+
+const managedCollection = await framer.getManagedCollection()
+const initialManagedCollectionFields: ManagedCollectionFields = await managedCollection.getFields()
+const initialFieldIds: FieldIds = new Set()
+
+interface FieldMappingProps {
+    collection: ManagedCollection
```

Comment:
Why are you loading the `managedCollection` outside of rendering and then also passing it into the component as a property?